src/Settings/Builders/Make.hs

```diff
@@ -13,5 +15,23 @@ makeBuilderArgs = do
     mconcat
         [ builder (Make gmpPath          ) ? pure ["MAKEFLAGS=-j" ++ t]
         , builder (Make libffiPath       ) ? pure ["MAKEFLAGS=-j" ++ t, "install"]
-        , builder (Make "testsuite/tests") ? pure ["THREADS=" ++ t, "fast"]
         ]
+
+validateBuilderArgs :: Args
+validateBuilderArgs = builder (Make "testsuite/tests") ? do
+    threads             <- shakeThreads <$> expr getShakeOptions
+    top                 <- expr topDirectory
+    compiler            <- expr $ fullpath ghc
+    checkPpr            <- expr $ fullpath checkPpr
+    checkApiAnnotations <- expr $ fullpath checkApiAnnotations
+    let t = show $ max 4 (threads - 2)
```

Comment:
Since the validation script will likely be the only running thread, we do not need to reduce parallelism like we do with other Make invocations. I suggest we use the `threads` settings as is.